gr_modtool bind fails with gcc-12 and -13 (e.g., Ubuntu 24)

### What happened?

While attempting to run gr_modtool bind <block> on a component of an OOT module the tool is failing with multiple c++ errors.  The same command works fine under gcc-11.

gnorbury@vaio:~/op25/op25/gr-op25$ gr_modtool bind decoder_bf
GNU Radio module name identified: op25
In file included from /tmp/tmp3yi889z4.h:1:
In file included from /home/gnorbury/op25.gr310_gcc12/op25/gr-op25/include/gnuradio/op25/decoder_bf.h:26:
In file included from /usr/local/include/gnuradio/block.h:14:
In file included from /usr/include/c++/12/memory:64:
In file included from /usr/include/c++/12/bits/allocator.h:46:
In file included from /usr/include/x86_64-linux-gnu/c++/12/bits/c++allocator.h:33:
/usr/include/c++/12/bits/new_allocator.h:153:6: error: call to '__builtin_operator_delete' selects
      non-usual deallocation function
            _GLIBCXX_OPERATOR_DELETE(_GLIBCXX_SIZED_DEALLOC(__p, __n),
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[snip]

In **blocktool/core/parseheader_generic.py**, changing the cflags from "-std=c++17 -fPIC" to "-std=c++17 -fPIC -fsized-deallocation" appears to correct the problem.

### System Information

OS: Ubuntu 22.10
GR Installation Method: native packages

Have repeated the test scenario using Ubuntu 22.04 using gcc-11 (works ok), then swapped to gcc-12 with everything else remaining unchanged and the bug makes an appearance.

### GNU Radio Version

3.10 (maint-3.10)

### Specific Version

v3.10.5.0-1-g480f600c

### Steps to Reproduce the Problem

Have been working with "https://github.com/boatbod/op25" OOT application
With the above repo cloned into the "op25" directory, you can run the gr_modtool as follows:

cd op25/op25/gr-op25
gr_modtool bind decoder_bf

Switching back and forward between gcc-11 and gcc-12 (using update_alternatives) yields either good or broken behavior.
Manually adding the "-fsized-deallocation" option to gr_modtool's cflags as described above eliminates the problem.

### Relevant log output

```shell
gnorbury@vaio:~/op25/op25/gr-op25$ gr_modtool bind decoder_bf
GNU Radio module name identified: op25
In file included from /tmp/tmptit6ewop.h:1:
In file included from /home/gnorbury/op25.gr310_gcc12/op25/gr-op25/include/gnuradio/op25/decoder_bf.h:26:
In file included from /usr/local/include/gnuradio/block.h:14:
In file included from /usr/include/c++/12/memory:64:
In file included from /usr/include/c++/12/bits/allocator.h:46:
In file included from /usr/include/x86_64-linux-gnu/c++/12/bits/c++allocator.h:33:
/usr/include/c++/12/bits/new_allocator.h:153:6: error: call to '__builtin_operator_delete' selects
      non-usual deallocation function
            _GLIBCXX_OPERATOR_DELETE(_GLIBCXX_SIZED_DEALLOC(__p, __n),
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/c++/12/bits/new_allocator.h:103:35: note: expanded from macro
      '_GLIBCXX_OPERATOR_DELETE'
# define _GLIBCXX_OPERATOR_DELETE __builtin_operator_delete
                                  ^
/usr/include/c++/12/bits/alloc_traits.h:496:13: note: in instantiation of member function
      'std::__new_allocator<char>::deallocate' requested here
      { __a.deallocate(__p, __n); }
            ^
/usr/include/c++/12/bits/basic_string.h:293:24: note: in instantiation of member function
      'std::allocator_traits<std::allocator<char>>::deallocate' requested here
      { _Alloc_traits::deallocate(_M_get_allocator(), _M_data(), __size + 1); }
                       ^
/usr/include/c++/12/bits/basic_string.h:287:4: note: in instantiation of member function
      'std::basic_string<char>::_M_destroy' requested here
          _M_destroy(_M_allocated_capacity);
          ^
/usr/include/c++/12/bits/basic_string.h:795:9: note: in instantiation of member function
      'std::basic_string<char>::_M_dispose' requested here
      { _M_dispose(); }
        ^
/usr/include/c++/12/bits/basic_string.h:4020:12: note: in instantiation of member function
      'std::basic_string<char>::~basic_string' requested here
    string __str(__neg + __len, '-');
           ^
/usr/include/c++/12/new:166:6: note: non-usual 'operator delete' declared here
void operator delete(void*, std::size_t, std::align_val_t)
~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /tmp/tmptit6ewop.h:1:
In file included from /home/gnorbury/op25.gr310_gcc12/op25/gr-op25/include/gnuradio/op25/decoder_bf.h:26:
In file included from /usr/local/include/gnuradio/block.h:14:
In file included from /usr/include/c++/12/memory:64:
In file included from /usr/include/c++/12/bits/allocator.h:46:
In file included from /usr/include/x86_64-linux-gnu/c++/12/bits/c++allocator.h:33:
/usr/include/c++/12/bits/new_allocator.h:158:2: error: call to '__builtin_operator_delete' selects
      non-usual deallocation function
        _GLIBCXX_OPERATOR_DELETE(_GLIBCXX_SIZED_DEALLOC(__p, __n));
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/c++/12/bits/new_allocator.h:103:35: note: expanded from macro
      '_GLIBCXX_OPERATOR_DELETE'
# define _GLIBCXX_OPERATOR_DELETE __builtin_operator_delete
                                  ^
/usr/include/c++/12/new:135:6: note: non-usual 'operator delete' declared here
void operator delete(void*, std::size_t) _GLIBCXX_USE_NOEXCEPT
~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /tmp/tmptit6ewop.h:1:
In file included from /home/gnorbury/op25.gr310_gcc12/op25/gr-op25/include/gnuradio/op25/decoder_bf.h:26:
In file included from /usr/local/include/gnuradio/block.h:14:
In file included from /usr/include/c++/12/memory:64:
In file included from /usr/include/c++/12/bits/allocator.h:46:
In file included from /usr/include/x86_64-linux-gnu/c++/12/bits/c++allocator.h:33:
/usr/include/c++/12/bits/new_allocator.h:153:6: error: call to '__builtin_operator_delete' selects
      non-usual deallocation function
            _GLIBCXX_OPERATOR_DELETE(_GLIBCXX_SIZED_DEALLOC(__p, __n),
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/c++/12/bits/new_allocator.h:103:35: note: expanded from macro
      '_GLIBCXX_OPERATOR_DELETE'
# define _GLIBCXX_OPERATOR_DELETE __builtin_operator_delete
                                  ^
/usr/include/c++/12/bits/alloc_traits.h:496:13: note: in instantiation of member function
      'std::__new_allocator<wchar_t>::deallocate' requested here
      { __a.deallocate(__p, __n); }
            ^
/usr/include/c++/12/bits/basic_string.h:293:24: note: in instantiation of member function
      'std::allocator_traits<std::allocator<wchar_t>>::deallocate' requested here
      { _Alloc_traits::deallocate(_M_get_allocator(), _M_data(), __size + 1); }
                       ^
/usr/include/c++/12/bits/basic_string.h:287:4: note: in instantiation of member function
      'std::basic_string<wchar_t>::_M_destroy' requested here
          _M_destroy(_M_allocated_capacity);
          ^
/usr/include/c++/12/bits/basic_string.tcc:238:44: note: in instantiation of member function
      'std::basic_string<wchar_t>::_M_dispose' requested here
          ~_Guard() { if (_M_guarded) _M_guarded->_M_dispose(); }
                                                  ^
/usr/include/c++/12/bits/basic_string.tcc:232:9: note: in instantiation of member function
      'std::basic_string<wchar_t>::_M_construct(wchar_t *, wchar_t *, std::forward_iterator_tag)::_Guard::~_Guard'
      requested here
        struct _Guard
               ^
/usr/include/c++/12/bits/basic_string.h:756:4: note: in instantiation of function template
      specialization 'std::basic_string<wchar_t>::_M_construct<wchar_t *>' requested here
          _M_construct(__beg, __end, std::__iterator_category(__beg));
          ^
/usr/include/c++/12/ext/string_conversions.h:115:14: note: in instantiation of function template
      specialization 'std::basic_string<wchar_t>::basic_string<wchar_t *, void>' requested here
      return _String(__s, __s + __len);
             ^
/usr/include/c++/12/bits/basic_string.h:4155:23: note: in instantiation of function template
      specialization '__gnu_cxx::__to_xstring<std::basic_string<wchar_t>, wchar_t>' requested here
  { return __gnu_cxx::__to_xstring<wstring>(&std::vswprintf, 4 * sizeof(int),
                      ^
/usr/include/c++/12/new:166:6: note: non-usual 'operator delete' declared here
void operator delete(void*, std::size_t, std::align_val_t)
~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /tmp/tmptit6ewop.h:1:
In file included from /home/gnorbury/op25.gr310_gcc12/op25/gr-op25/include/gnuradio/op25/decoder_bf.h:26:
In file included from /usr/local/include/gnuradio/block.h:14:
In file included from /usr/include/c++/12/memory:64:
In file included from /usr/include/c++/12/bits/allocator.h:46:
In file included from /usr/include/x86_64-linux-gnu/c++/12/bits/c++allocator.h:33:
/usr/include/c++/12/bits/new_allocator.h:158:2: error: call to '__builtin_operator_delete' selects
      non-usual deallocation function
        _GLIBCXX_OPERATOR_DELETE(_GLIBCXX_SIZED_DEALLOC(__p, __n));
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/c++/12/bits/new_allocator.h:103:35: note: expanded from macro
      '_GLIBCXX_OPERATOR_DELETE'
# define _GLIBCXX_OPERATOR_DELETE __builtin_operator_delete
                                  ^
/usr/include/c++/12/new:135:6: note: non-usual 'operator delete' declared here
void operator delete(void*, std::size_t) _GLIBCXX_USE_NOEXCEPT
~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /tmp/tmptit6ewop.h:1:
In file included from /home/gnorbury/op25.gr310_gcc12/op25/gr-op25/include/gnuradio/op25/decoder_bf.h:26:
In file included from /usr/local/include/gnuradio/block.h:14:
In file included from /usr/include/c++/12/memory:64:
In file included from /usr/include/c++/12/bits/allocator.h:46:
In file included from /usr/include/x86_64-linux-gnu/c++/12/bits/c++allocator.h:33:
/usr/include/c++/12/bits/new_allocator.h:153:6: error: call to '__builtin_operator_delete' selects
      non-usual deallocation function
            _GLIBCXX_OPERATOR_DELETE(_GLIBCXX_SIZED_DEALLOC(__p, __n),
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/c++/12/bits/new_allocator.h:103:35: note: expanded from macro
      '_GLIBCXX_OPERATOR_DELETE'
# define _GLIBCXX_OPERATOR_DELETE __builtin_operator_delete
                                  ^
/usr/include/c++/12/bits/alloc_traits.h:496:13: note: in instantiation of member function
      'std::__new_allocator<char16_t>::deallocate' requested here
      { __a.deallocate(__p, __n); }
            ^
/usr/include/c++/12/bits/basic_string.h:293:24: note: in instantiation of member function
      'std::allocator_traits<std::allocator<char16_t>>::deallocate' requested here
      { _Alloc_traits::deallocate(_M_get_allocator(), _M_data(), __size + 1); }
                       ^
/usr/include/c++/12/bits/basic_string.h:287:4: note: in instantiation of member function
      'std::basic_string<char16_t>::_M_destroy' requested here
          _M_destroy(_M_allocated_capacity);
          ^
/usr/include/c++/12/bits/basic_string.tcc:238:44: note: in instantiation of member function
      'std::basic_string<char16_t>::_M_dispose' requested here
          ~_Guard() { if (_M_guarded) _M_guarded->_M_dispose(); }
                                                  ^
/usr/include/c++/12/bits/basic_string.tcc:232:9: note: in instantiation of member function
      'std::basic_string<char16_t>::_M_construct(const char16_t *, const char16_t *,
      std::forward_iterator_tag)::_Guard::~_Guard' requested here
        struct _Guard
               ^
/usr/include/c++/12/bits/basic_string.h:620:2: note: in instantiation of function template
      specialization 'std::basic_string<char16_t>::_M_construct<const char16_t *>' requested here
        _M_construct(__s, __s + __n, std::forward_iterator_tag());
        ^
/usr/include/c++/12/bits/basic_string.h:4347:14: note: in instantiation of member function
      'std::basic_string<char16_t>::basic_string' requested here
    { return basic_string<char16_t>{__str, __len}; }
             ^
/usr/include/c++/12/new:166:6: note: non-usual 'operator delete' declared here
void operator delete(void*, std::size_t, std::align_val_t)
~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /tmp/tmptit6ewop.h:1:
In file included from /home/gnorbury/op25.gr310_gcc12/op25/gr-op25/include/gnuradio/op25/decoder_bf.h:26:
In file included from /usr/local/include/gnuradio/block.h:14:
In file included from /usr/include/c++/12/memory:64:
In file included from /usr/include/c++/12/bits/allocator.h:46:
In file included from /usr/include/x86_64-linux-gnu/c++/12/bits/c++allocator.h:33:
/usr/include/c++/12/bits/new_allocator.h:158:2: error: call to '__builtin_operator_delete' selects
      non-usual deallocation function
        _GLIBCXX_OPERATOR_DELETE(_GLIBCXX_SIZED_DEALLOC(__p, __n));
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/c++/12/bits/new_allocator.h:103:35: note: expanded from macro
      '_GLIBCXX_OPERATOR_DELETE'
# define _GLIBCXX_OPERATOR_DELETE __builtin_operator_delete
                                  ^
/usr/include/c++/12/new:135:6: note: non-usual 'operator delete' declared here
void operator delete(void*, std::size_t) _GLIBCXX_USE_NOEXCEPT
~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /tmp/tmptit6ewop.h:1:
In file included from /home/gnorbury/op25.gr310_gcc12/op25/gr-op25/include/gnuradio/op25/decoder_bf.h:26:
In file included from /usr/local/include/gnuradio/block.h:14:
In file included from /usr/include/c++/12/memory:64:
In file included from /usr/include/c++/12/bits/allocator.h:46:
In file included from /usr/include/x86_64-linux-gnu/c++/12/bits/c++allocator.h:33:
/usr/include/c++/12/bits/new_allocator.h:153:6: error: call to '__builtin_operator_delete' selects
      non-usual deallocation function
            _GLIBCXX_OPERATOR_DELETE(_GLIBCXX_SIZED_DEALLOC(__p, __n),
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/c++/12/bits/new_allocator.h:103:35: note: expanded from macro
      '_GLIBCXX_OPERATOR_DELETE'
# define _GLIBCXX_OPERATOR_DELETE __builtin_operator_delete
                                  ^
/usr/include/c++/12/bits/alloc_traits.h:496:13: note: in instantiation of member function
      'std::__new_allocator<char32_t>::deallocate' requested here
      { __a.deallocate(__p, __n); }
            ^
/usr/include/c++/12/bits/basic_string.h:293:24: note: in instantiation of member function
      'std::allocator_traits<std::allocator<char32_t>>::deallocate' requested here
      { _Alloc_traits::deallocate(_M_get_allocator(), _M_data(), __size + 1); }
                       ^
/usr/include/c++/12/bits/basic_string.h:287:4: note: in instantiation of member function
      'std::basic_string<char32_t>::_M_destroy' requested here
          _M_destroy(_M_allocated_capacity);
          ^
/usr/include/c++/12/bits/basic_string.tcc:238:44: note: in instantiation of member function
      'std::basic_string<char32_t>::_M_dispose' requested here
          ~_Guard() { if (_M_guarded) _M_guarded->_M_dispose(); }
                                                  ^
/usr/include/c++/12/bits/basic_string.tcc:232:9: note: in instantiation of member function
      'std::basic_string<char32_t>::_M_construct(const char32_t *, const char32_t *,
      std::forward_iterator_tag)::_Guard::~_Guard' requested here
        struct _Guard
               ^
/usr/include/c++/12/bits/basic_string.h:620:2: note: in instantiation of function template
      specialization 'std::basic_string<char32_t>::_M_construct<const char32_t *>' requested here
        _M_construct(__s, __s + __n, std::forward_iterator_tag());
        ^
/usr/include/c++/12/bits/basic_string.h:4352:14: note: in instantiation of member function
      'std::basic_string<char32_t>::basic_string' requested here
    { return basic_string<char32_t>{__str, __len}; }
             ^
/usr/include/c++/12/new:166:6: note: non-usual 'operator delete' declared here
void operator delete(void*, std::size_t, std::align_val_t)
~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /tmp/tmptit6ewop.h:1:
In file included from /home/gnorbury/op25.gr310_gcc12/op25/gr-op25/include/gnuradio/op25/decoder_bf.h:26:
In file included from /usr/local/include/gnuradio/block.h:14:
In file included from /usr/include/c++/12/memory:64:
In file included from /usr/include/c++/12/bits/allocator.h:46:
In file included from /usr/include/x86_64-linux-gnu/c++/12/bits/c++allocator.h:33:
/usr/include/c++/12/bits/new_allocator.h:158:2: error: call to '__builtin_operator_delete' selects
      non-usual deallocation function
        _GLIBCXX_OPERATOR_DELETE(_GLIBCXX_SIZED_DEALLOC(__p, __n));
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/c++/12/bits/new_allocator.h:103:35: note: expanded from macro
      '_GLIBCXX_OPERATOR_DELETE'
# define _GLIBCXX_OPERATOR_DELETE __builtin_operator_delete
                                  ^
/usr/include/c++/12/new:135:6: note: non-usual 'operator delete' declared here
void operator delete(void*, std::size_t) _GLIBCXX_USE_NOEXCEPT
~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /tmp/tmptit6ewop.h:1:
In file included from /home/gnorbury/op25.gr310_gcc12/op25/gr-op25/include/gnuradio/op25/decoder_bf.h:26:
In file included from /usr/local/include/gnuradio/block.h:14:
In file included from /usr/include/c++/12/memory:64:
In file included from /usr/include/c++/12/bits/allocator.h:46:
In file included from /usr/include/x86_64-linux-gnu/c++/12/bits/c++allocator.h:33:
/usr/include/c++/12/bits/new_allocator.h:153:6: error: call to '__builtin_operator_delete' selects
      non-usual deallocation function
            _GLIBCXX_OPERATOR_DELETE(_GLIBCXX_SIZED_DEALLOC(__p, __n),
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/c++/12/bits/new_allocator.h:103:35: note: expanded from macro
      '_GLIBCXX_OPERATOR_DELETE'
# define _GLIBCXX_OPERATOR_DELETE __builtin_operator_delete
                                  ^
/usr/include/c++/12/bits/alloc_traits.h:496:13: note: in instantiation of member function
      'std::__new_allocator<std::_Rb_tree_node<std::pair<const boost::system::error_category *const,
      std::unique_ptr<boost::system::detail::std_category>>>>::deallocate' requested here
      { __a.deallocate(__p, __n); }
            ^
/usr/include/c++/12/bits/stl_tree.h:565:24: note: in instantiation of member function
      'std::allocator_traits<std::allocator<std::_Rb_tree_node<std::pair<const boost::system::error_category *const,
      std::unique_ptr<boost::system::detail::std_category>>>>>::deallocate' requested here
      { _Alloc_traits::deallocate(_M_get_Node_allocator(), __p, 1); }
                       ^
/usr/include/c++/12/bits/stl_tree.h:632:2: note: in instantiation of member function
      'std::_Rb_tree<const boost::system::error_category *, std::pair<const boost::system::error_category *const,
      std::unique_ptr<boost::system::detail::std_category>>, std::_Select1st<std::pair<const
      boost::system::error_category *const, std::unique_ptr<boost::system::detail::std_category>>>,
      boost::system::detail::cat_ptr_less>::_M_put_node' requested here
        _M_put_node(__p);
        ^
/usr/include/c++/12/bits/stl_tree.h:1937:4: note: in instantiation of member function
      'std::_Rb_tree<const boost::system::error_category *, std::pair<const boost::system::error_category *const,
      std::unique_ptr<boost::system::detail::std_category>>, std::_Select1st<std::pair<const
      boost::system::error_category *const, std::unique_ptr<boost::system::detail::std_category>>>,
      boost::system::detail::cat_ptr_less>::_M_drop_node' requested here
          _M_drop_node(__x);
          ^
/usr/include/c++/12/bits/stl_tree.h:984:9: note: in instantiation of member function
      'std::_Rb_tree<const boost::system::error_category *, std::pair<const boost::system::error_category *const,
      std::unique_ptr<boost::system::detail::std_category>>, std::_Select1st<std::pair<const
      boost::system::error_category *const, std::unique_ptr<boost::system::detail::std_category>>>,
      boost::system::detail::cat_ptr_less>::_M_erase' requested here
      { _M_erase(_M_begin()); }
        ^
/usr/include/c++/12/bits/stl_map.h:195:7: note: in instantiation of member function
      'std::_Rb_tree<const boost::system::error_category *, std::pair<const boost::system::error_category *const,
      std::unique_ptr<boost::system::detail::std_category>>, std::_Select1st<std::pair<const
      boost::system::error_category *const, std::unique_ptr<boost::system::detail::std_category>>>,
      boost::system::detail::cat_ptr_less>::~_Rb_tree' requested here
      map() = default;
      ^
/usr/include/c++/12/new:166:6: note: non-usual 'operator delete' declared here
void operator delete(void*, std::size_t, std::align_val_t)
~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /tmp/tmptit6ewop.h:1:
In file included from /home/gnorbury/op25.gr310_gcc12/op25/gr-op25/include/gnuradio/op25/decoder_bf.h:26:
In file included from /usr/local/include/gnuradio/block.h:14:
In file included from /usr/include/c++/12/memory:64:
In file included from /usr/include/c++/12/bits/allocator.h:46:
In file included from /usr/include/x86_64-linux-gnu/c++/12/bits/c++allocator.h:33:
/usr/include/c++/12/bits/new_allocator.h:158:2: error: call to '__builtin_operator_delete' selects
      non-usual deallocation function
        _GLIBCXX_OPERATOR_DELETE(_GLIBCXX_SIZED_DEALLOC(__p, __n));
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/c++/12/bits/new_allocator.h:103:35: note: expanded from macro
      '_GLIBCXX_OPERATOR_DELETE'
# define _GLIBCXX_OPERATOR_DELETE __builtin_operator_delete
                                  ^
/usr/include/c++/12/new:135:6: note: non-usual 'operator delete' declared here
void operator delete(void*, std::size_t) _GLIBCXX_USE_NOEXCEPT
~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /tmp/tmptit6ewop.h:1:
In file included from /home/gnorbury/op25.gr310_gcc12/op25/gr-op25/include/gnuradio/op25/decoder_bf.h:26:
In file included from /usr/local/include/gnuradio/block.h:14:
In file included from /usr/include/c++/12/memory:64:
In file included from /usr/include/c++/12/bits/allocator.h:46:
In file included from /usr/include/x86_64-linux-gnu/c++/12/bits/c++allocator.h:33:
/usr/include/c++/12/bits/new_allocator.h:153:6: error: call to '__builtin_operator_delete' selects
      non-usual deallocation function
            _GLIBCXX_OPERATOR_DELETE(_GLIBCXX_SIZED_DEALLOC(__p, __n),
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/c++/12/bits/new_allocator.h:103:35: note: expanded from macro
      '_GLIBCXX_OPERATOR_DELETE'
# define _GLIBCXX_OPERATOR_DELETE __builtin_operator_delete
                                  ^
/usr/include/c++/12/bits/alloc_traits.h:496:13: note: in instantiation of member function
      'std::__new_allocator<std::pair<boost::condition_variable *, boost::mutex *>>::deallocate' requested here
      { __a.deallocate(__p, __n); }
            ^
/usr/include/c++/12/bits/stl_vector.h:387:9: note: in instantiation of member function
      'std::allocator_traits<std::allocator<std::pair<boost::condition_variable *, boost::mutex *>>>::deallocate'
      requested here
          _Tr::deallocate(_M_impl, __p, __n);
               ^
/usr/include/c++/12/bits/stl_vector.h:366:2: note: in instantiation of member function
      'std::_Vector_base<std::pair<boost::condition_variable *, boost::mutex *>,
      std::allocator<std::pair<boost::condition_variable *, boost::mutex *>>>::_M_deallocate' requested here
        _M_deallocate(_M_impl._M_start,
        ^
/usr/include/c++/12/bits/stl_vector.h:526:7: note: in instantiation of member function
      'std::_Vector_base<std::pair<boost::condition_variable *, boost::mutex *>,
      std::allocator<std::pair<boost::condition_variable *, boost::mutex *>>>::~_Vector_base' requested here
      vector() = default;
      ^
/usr/include/c++/12/new:166:6: note: non-usual 'operator delete' declared here
void operator delete(void*, std::size_t, std::align_val_t)
~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /tmp/tmptit6ewop.h:1:
In file included from /home/gnorbury/op25.gr310_gcc12/op25/gr-op25/include/gnuradio/op25/decoder_bf.h:26:
In file included from /usr/local/include/gnuradio/block.h:14:
In file included from /usr/include/c++/12/memory:64:
In file included from /usr/include/c++/12/bits/allocator.h:46:
In file included from /usr/include/x86_64-linux-gnu/c++/12/bits/c++allocator.h:33:
/usr/include/c++/12/bits/new_allocator.h:158:2: error: call to '__builtin_operator_delete' selects
      non-usual deallocation function
        _GLIBCXX_OPERATOR_DELETE(_GLIBCXX_SIZED_DEALLOC(__p, __n));
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/c++/12/bits/new_allocator.h:103:35: note: expanded from macro
      '_GLIBCXX_OPERATOR_DELETE'
# define _GLIBCXX_OPERATOR_DELETE __builtin_operator_delete
                                  ^
/usr/include/c++/12/new:135:6: note: non-usual 'operator delete' declared here
void operator delete(void*, std::size_t) _GLIBCXX_USE_NOEXCEPT
~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /tmp/tmptit6ewop.h:1:
In file included from /home/gnorbury/op25.gr310_gcc12/op25/gr-op25/include/gnuradio/op25/decoder_bf.h:26:
In file included from /usr/local/include/gnuradio/block.h:14:
In file included from /usr/include/c++/12/memory:64:
In file included from /usr/include/c++/12/bits/allocator.h:46:
In file included from /usr/include/x86_64-linux-gnu/c++/12/bits/c++allocator.h:33:
/usr/include/c++/12/bits/new_allocator.h:153:6: error: call to '__builtin_operator_delete' selects
      non-usual deallocation function
            _GLIBCXX_OPERATOR_DELETE(_GLIBCXX_SIZED_DEALLOC(__p, __n),
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/c++/12/bits/new_allocator.h:103:35: note: expanded from macro
      '_GLIBCXX_OPERATOR_DELETE'
# define _GLIBCXX_OPERATOR_DELETE __builtin_operator_delete
                                  ^
/usr/include/c++/12/bits/alloc_traits.h:496:13: note: in instantiation of member function
      'std::__new_allocator<boost::shared_ptr<boost::detail::shared_state_base>>::deallocate' requested here
      { __a.deallocate(__p, __n); }
            ^
/usr/include/c++/12/bits/stl_vector.h:387:9: note: in instantiation of member function
      'std::allocator_traits<std::allocator<boost::shared_ptr<boost::detail::shared_state_base>>>::deallocate' requested
      here
          _Tr::deallocate(_M_impl, __p, __n);
               ^
/usr/include/c++/12/bits/stl_vector.h:366:2: note: in instantiation of member function
      'std::_Vector_base<boost::shared_ptr<boost::detail::shared_state_base>,
      std::allocator<boost::shared_ptr<boost::detail::shared_state_base>>>::_M_deallocate' requested here
        _M_deallocate(_M_impl._M_start,
        ^
/usr/include/c++/12/bits/stl_vector.h:526:7: note: in instantiation of member function
      'std::_Vector_base<boost::shared_ptr<boost::detail::shared_state_base>,
      std::allocator<boost::shared_ptr<boost::detail::shared_state_base>>>::~_Vector_base' requested here
      vector() = default;
      ^
/usr/include/c++/12/new:166:6: note: non-usual 'operator delete' declared here
void operator delete(void*, std::size_t, std::align_val_t)
~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /tmp/tmptit6ewop.h:1:
In file included from /home/gnorbury/op25.gr310_gcc12/op25/gr-op25/include/gnuradio/op25/decoder_bf.h:26:
In file included from /usr/local/include/gnuradio/block.h:14:
In file included from /usr/include/c++/12/memory:64:
In file included from /usr/include/c++/12/bits/allocator.h:46:
In file included from /usr/include/x86_64-linux-gnu/c++/12/bits/c++allocator.h:33:
/usr/include/c++/12/bits/new_allocator.h:158:2: error: call to '__builtin_operator_delete' selects
      non-usual deallocation function
        _GLIBCXX_OPERATOR_DELETE(_GLIBCXX_SIZED_DEALLOC(__p, __n));
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/c++/12/bits/new_allocator.h:103:35: note: expanded from macro
      '_GLIBCXX_OPERATOR_DELETE'
# define _GLIBCXX_OPERATOR_DELETE __builtin_operator_delete
                                  ^
/usr/include/c++/12/new:135:6: note: non-usual 'operator delete' declared here
void operator delete(void*, std::size_t) _GLIBCXX_USE_NOEXCEPT
~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /tmp/tmptit6ewop.h:1:
In file included from /home/gnorbury/op25.gr310_gcc12/op25/gr-op25/include/gnuradio/op25/decoder_bf.h:26:
In file included from /usr/local/include/gnuradio/block.h:14:
In file included from /usr/include/c++/12/memory:64:
In file included from /usr/include/c++/12/bits/allocator.h:46:
In file included from /usr/include/x86_64-linux-gnu/c++/12/bits/c++allocator.h:33:
/usr/include/c++/12/bits/new_allocator.h:153:6: error: call to '__builtin_operator_delete' selects
      non-usual deallocation function
            _GLIBCXX_OPERATOR_DELETE(_GLIBCXX_SIZED_DEALLOC(__p, __n),
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/c++/12/bits/new_allocator.h:103:35: note: expanded from macro
      '_GLIBCXX_OPERATOR_DELETE'
# define _GLIBCXX_OPERATOR_DELETE __builtin_operator_delete
                                  ^
/usr/include/c++/12/bits/alloc_traits.h:496:13: note: in instantiation of member function
      'std::__new_allocator<std::_Rb_tree_node<std::pair<const void *const, boost::detail::tss_data_node>>>::deallocate'
      requested here
      { __a.deallocate(__p, __n); }
            ^
/usr/include/c++/12/bits/stl_tree.h:565:24: note: in instantiation of member function
      'std::allocator_traits<std::allocator<std::_Rb_tree_node<std::pair<const void *const,
      boost::detail::tss_data_node>>>>::deallocate' requested here
      { _Alloc_traits::deallocate(_M_get_Node_allocator(), __p, 1); }
                       ^
/usr/include/c++/12/bits/stl_tree.h:632:2: note: in instantiation of member function
      'std::_Rb_tree<const void *, std::pair<const void *const, boost::detail::tss_data_node>,
      std::_Select1st<std::pair<const void *const, boost::detail::tss_data_node>>, std::less<const void
      *>>::_M_put_node' requested here
        _M_put_node(__p);
        ^
/usr/include/c++/12/bits/stl_tree.h:1937:4: note: in instantiation of member function
      'std::_Rb_tree<const void *, std::pair<const void *const, boost::detail::tss_data_node>,
      std::_Select1st<std::pair<const void *const, boost::detail::tss_data_node>>, std::less<const void
      *>>::_M_drop_node' requested here
          _M_drop_node(__x);
          ^
/usr/include/c++/12/bits/stl_tree.h:984:9: note: in instantiation of member function
      'std::_Rb_tree<const void *, std::pair<const void *const, boost::detail::tss_data_node>,
      std::_Select1st<std::pair<const void *const, boost::detail::tss_data_node>>, std::less<const void *>>::_M_erase'
      requested here
      { _M_erase(_M_begin()); }
        ^
/usr/include/c++/12/bits/stl_map.h:195:7: note: in instantiation of member function
      'std::_Rb_tree<const void *, std::pair<const void *const, boost::detail::tss_data_node>,
      std::_Select1st<std::pair<const void *const, boost::detail::tss_data_node>>, std::less<const void *>>::~_Rb_tree'
      requested here
      map() = default;
      ^
/usr/include/c++/12/new:166:6: note: non-usual 'operator delete' declared here
void operator delete(void*, std::size_t, std::align_val_t)
~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /tmp/tmptit6ewop.h:1:
In file included from /home/gnorbury/op25.gr310_gcc12/op25/gr-op25/include/gnuradio/op25/decoder_bf.h:26:
In file included from /usr/local/include/gnuradio/block.h:14:
In file included from /usr/include/c++/12/memory:64:
In file included from /usr/include/c++/12/bits/allocator.h:46:
In file included from /usr/include/x86_64-linux-gnu/c++/12/bits/c++allocator.h:33:
/usr/include/c++/12/bits/new_allocator.h:158:2: error: call to '__builtin_operator_delete' selects
      non-usual deallocation function
        _GLIBCXX_OPERATOR_DELETE(_GLIBCXX_SIZED_DEALLOC(__p, __n));
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/c++/12/bits/new_allocator.h:103:35: note: expanded from macro
      '_GLIBCXX_OPERATOR_DELETE'
# define _GLIBCXX_OPERATOR_DELETE __builtin_operator_delete
                                  ^
/usr/include/c++/12/new:135:6: note: non-usual 'operator delete' declared here
void operator delete(void*, std::size_t) _GLIBCXX_USE_NOEXCEPT
~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /tmp/tmptit6ewop.h:1:
In file included from /home/gnorbury/op25.gr310_gcc12/op25/gr-op25/include/gnuradio/op25/decoder_bf.h:26:
In file included from /usr/local/include/gnuradio/block.h:14:
In file included from /usr/include/c++/12/memory:64:
In file included from /usr/include/c++/12/bits/allocator.h:46:
In file included from /usr/include/x86_64-linux-gnu/c++/12/bits/c++allocator.h:33:
/usr/include/c++/12/bits/new_allocator.h:153:6: error: call to '__builtin_operator_delete' selects
      non-usual deallocation function
            _GLIBCXX_OPERATOR_DELETE(_GLIBCXX_SIZED_DEALLOC(__p, __n),
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/c++/12/bits/new_allocator.h:103:35: note: expanded from macro
      '_GLIBCXX_OPERATOR_DELETE'
# define _GLIBCXX_OPERATOR_DELETE __builtin_operator_delete
                                  ^
/usr/include/c++/12/bits/alloc_traits.h:496:13: note: in instantiation of member function
      'std::__new_allocator<std::_List_node<boost::thread *>>::deallocate' requested here
      { __a.deallocate(__p, __n); }
            ^
/usr/include/c++/12/bits/stl_list.h:522:29: note: in instantiation of member function
      'std::allocator_traits<std::allocator<std::_List_node<boost::thread *>>>::deallocate' requested here
      { _Node_alloc_traits::deallocate(_M_impl, __p, 1); }
                            ^
/usr/include/c++/12/bits/list.tcc:81:4: note: in instantiation of member function
      'std::_List_base<boost::thread *, std::allocator<boost::thread *>>::_M_put_node' requested here
          _M_put_node(__tmp);
          ^
/usr/include/c++/12/bits/stl_list.h:575:9: note: in instantiation of member function
      'std::_List_base<boost::thread *, std::allocator<boost::thread *>>::_M_clear' requested here
      { _M_clear(); }
        ^
/usr/include/c++/12/bits/stl_list.h:749:7: note: in instantiation of member function
      'std::_List_base<boost::thread *, std::allocator<boost::thread *>>::~_List_base' requested here
      list() = default;
      ^
/usr/include/c++/12/new:166:6: note: non-usual 'operator delete' declared here
void operator delete(void*, std::size_t, std::align_val_t)
~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /tmp/tmptit6ewop.h:1:
In file included from /home/gnorbury/op25.gr310_gcc12/op25/gr-op25/include/gnuradio/op25/decoder_bf.h:26:
In file included from /usr/local/include/gnuradio/block.h:14:
In file included from /usr/include/c++/12/memory:64:
In file included from /usr/include/c++/12/bits/allocator.h:46:
In file included from /usr/include/x86_64-linux-gnu/c++/12/bits/c++allocator.h:33:
/usr/include/c++/12/bits/new_allocator.h:158:2: error: call to '__builtin_operator_delete' selects
      non-usual deallocation function
        _GLIBCXX_OPERATOR_DELETE(_GLIBCXX_SIZED_DEALLOC(__p, __n));
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/c++/12/bits/new_allocator.h:103:35: note: expanded from macro
      '_GLIBCXX_OPERATOR_DELETE'
# define _GLIBCXX_OPERATOR_DELETE __builtin_operator_delete
                                  ^
/usr/include/c++/12/new:135:6: note: non-usual 'operator delete' declared here
void operator delete(void*, std::size_t) _GLIBCXX_USE_NOEXCEPT
~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /tmp/tmptit6ewop.h:1:
In file included from /home/gnorbury/op25.gr310_gcc12/op25/gr-op25/include/gnuradio/op25/decoder_bf.h:26:
In file included from /usr/local/include/gnuradio/block.h:14:
In file included from /usr/include/c++/12/memory:64:
In file included from /usr/include/c++/12/bits/allocator.h:46:
In file included from /usr/include/x86_64-linux-gnu/c++/12/bits/c++allocator.h:33:
/usr/include/c++/12/bits/new_allocator.h:153:6: error: call to '__builtin_operator_delete' selects
      non-usual deallocation function
            _GLIBCXX_OPERATOR_DELETE(_GLIBCXX_SIZED_DEALLOC(__p, __n),
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/c++/12/bits/new_allocator.h:103:35: note: expanded from macro
      '_GLIBCXX_OPERATOR_DELETE'
# define _GLIBCXX_OPERATOR_DELETE __builtin_operator_delete
                                  ^
/usr/include/c++/12/bits/alloc_traits.h:496:13: note: in instantiation of member function
      'std::__new_allocator<long>::deallocate' requested here
      { __a.deallocate(__p, __n); }
            ^
/usr/include/c++/12/bits/stl_vector.h:387:9: note: in instantiation of member function
      'std::allocator_traits<std::allocator<long>>::deallocate' requested here
          _Tr::deallocate(_M_impl, __p, __n);
               ^
/usr/include/c++/12/bits/stl_vector.h:366:2: note: in instantiation of member function
      'std::_Vector_base<long, std::allocator<long>>::_M_deallocate' requested here
        _M_deallocate(_M_impl._M_start,
        ^
/usr/include/c++/12/bits/stl_vector.h:526:7: note: in instantiation of member function
      'std::_Vector_base<long, std::allocator<long>>::~_Vector_base' requested here
      vector() = default;
      ^
/usr/include/c++/12/new:166:6: note: non-usual 'operator delete' declared here
void operator delete(void*, std::size_t, std::align_val_t)
~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.
Error occurred while running CASTXML xml file does not exist
gnorbury@vaio:~/op25/op25/gr-op25$
```

--- a/gr-utils/blocktool/core/parseheader.py
+++ b/gr-utils/blocktool/core/parseheader.py
@@ -101,7 +101,9 @@ class BlockHeaderParser(BlockTool):
             include_paths=self.include_paths,
             compiler='gcc',
             define_symbols=['BOOST_ATOMIC_DETAIL_EXTRA_BACKEND_GENERIC'],
-            cflags='-std=c++11')
+            # See the comment parseheader_generic: -fsized-deallocation
+            # helps what seems to be a compiler bug (?).
+            cflags='-std=c++17 -fPIC -fsized-deallocation')
         decls = parser.parse(
             [self.target_file], xml_generator_config)
         global_namespace = declarations.get_global_namespace(decls)

--- a/gr-utils/blocktool/core/parseheader_generic.py
+++ b/gr-utils/blocktool/core/parseheader_generic.py
@@ -48,7 +48,7 @@ class GenericHeaderParser(BlockTool):
         self.parsed_data = {}
         self.addcomments = blocktool_comments
         self.define_symbols = ('BOOST_ATOMIC_DETAIL_EXTRA_BACKEND_GENERIC', 'DISABLE_LOGGER_H')
-        if (define_symbols):
+        if define_symbols:
             self.define_symbols += define_symbols
         self.include_paths = None
         if (include_paths):

--- a/gr-utils/blocktool/core/parseheader_generic.py
+++ b/gr-utils/blocktool/core/parseheader_generic.py
@@ -48,7 +48,7 @@ class GenericHeaderParser(BlockTool):
         self.parsed_data = {}
         self.addcomments = blocktool_comments
         self.define_symbols = ('BOOST_ATOMIC_DETAIL_EXTRA_BACKEND_GENERIC', 'DISABLE_LOGGER_H')
-        if(define_symbols):
+        if (define_symbols):
             self.define_symbols += define_symbols
         self.include_paths = None
         if (include_paths):
@@ -318,10 +318,17 @@ class GenericHeaderParser(BlockTool):
                 xml_generator_path=generator_path,
                 xml_generator=generator_name,
                 include_paths=self.include_paths,
-                compiler='gcc',
+                compiler='g++',
                 undefine_symbols=['__PIE__'],
                 define_symbols=self.define_symbols,
-                cflags='-std=c++17 -fPIC')
+                # There seems to be a bug where on some platforms, GCC fails because it finds the
+                # _GLIBCXX_OPERATOR_DELETE and considers it a "non-usual deallocation function".
+                # Even if that is inherent to C++17. Anyways, in
+                # https://github.com/gnuradio/gnuradio/issues/6477 user boatbod convinces us that
+                # adding -fsized_deallocations helps this. Neither we nor the GCC mailing lists
+                # are perfectly sure why that is, but it does help. Since this is just for binding,
+                # none of the thus-generated code ends up anywhere, so here we go:
+                cflags='-std=c++17 -fPIC -fsized-deallocation')
             decls = parser.parse(
                 [self.target_file], xml_generator_config,
                 compilation_mode=parser.COMPILATION_MODE.ALL_AT_ONCE)

